cucumber/kubernetes/features/authenticate_with_annotations.feature

```diff
@@ -0,0 +1,27 @@
+Feature: An authorized client can authenticate as a permitted role with the host's
+  application identity defined in its annotations
+
+  Scenario: Authenticate as a Pod.
+    Given I login to pod matching "app=inventory-pod" to authn-k8s as "test-app-pod" with prefix "host/some-policy"
+    Then I can authenticate pod matching "pod/inventory-pod" with authn-k8s as "test-app-pod" with prefix "host/some-policy"
+
+  Scenario: Authenticate as a Namespace.
+    Given I can login to pod matching "app=inventory-pod" to authn-k8s as "test-app-namespace" with prefix "host/some-policy"
+    Then I can authenticate pod matching "pod/inventory-pod" with authn-k8s as "test-app-namespace" with prefix "host/some-policy"
+
+  Scenario: Authenticate as a ServiceAccount.
+    Given I can login to pod matching "app=inventory-pod" to authn-k8s as "test-app-service-account" with prefix "host/some-policy"
+    Then I can authenticate pod matching "pod/inventory-pod" with authn-k8s as "test-app-service-account" with prefix "host/some-policy"
+
+  Scenario: Authenticate as a Deployment.
+    Given I can login to pod matching "app=inventory-deployment" to authn-k8s as "test-app-deployment" with prefix "host/some-policy"
+    Then I can authenticate pod matching "pod/inventory-deployment" with authn-k8s as "test-app-deployment" with prefix "host/some-policy"
+
+  Scenario: Authenticate as a StatefulSet.
+    Given I can login to pod matching "app=inventory-stateful" to authn-k8s as "test-app-stateful-set" with prefix "host/some-policy"
+    Then I can authenticate pod matching "pod/inventory-stateful" with authn-k8s as "test-app-stateful-set" with prefix "host/some-policy"
+
+  @k8s_skip
```

Comment:
Not sure how familiar we expect devs to be with k8s/openshift differences so might be helpful here to add a comment that we are skipping b.c this test is for openshift only

Reply:
the tag `@k8s_skip` tells the story